### PR TITLE
Black Market code, Revisited.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

This PR updates and refreshes the whole of black market code for improved usability as well as to better sell the backbone of the content behind the blackmarket in-game.

For starters, the datums for the black market were designed around not being specific to the black market. Reading the code, it was intended to allow for multiple blackmarket_markets to be added after it's original inclusion, which was passed up as a result of what I'd guess is a branding issue, as every datum associated with the blackmarket was labeled... as for the black market, nothing else.

So to begin I've renamed most of the backend of the blackmarket code to just market instead, datum/market, datum/market_order, datum/market_item, datum/market_uplink(/blackmarket). The works.

Next, QOL change to how blackmarket uplinks were implemented: Now, instead of having to manually load credits into a black market uplink by hand, then choose to buy things using the uplink, they instead just draw from the user's ID card, checks for a bank account, and purchases through that, with quick inputs added when purchases are successful and warnings when a purchase cannot be made.

Lastly, code change. In an old economy PR of mine I standardized purchased cargo items to use the CARGO_CRATE_VALUE define, and for vendible items to use paycheck defines instead. In that PR I rebalanced quite a bit of prices as a result, but this got passed up when that happened. I'll leave the balancing for another time then, but this updates the code of market_item datums to use CARGO_CRATE_VALUE for their upper and lower cost ranges to maintain that standard. 

## Why It's Good For The Game

Honestly, I've pretty much never used the black market in-game. As the weirdo who has tried to scrape every ounce of joy out of the in-game economy, it just never really clicked with me, so I decided to give it a quick glance this week as I was finally getting a chance to look at some SS13 code for the first time, and I was impressed by how decently though out the Blackmarket code was put together. That said, there are elements of the it that needed some basic updating and I thought I'd help modernize the code just a smidge while I'm strolling through the neighborhood.

## Changelog

:cl:
qol: Blackmarket uplinks now draw from the player's ID card, not from an internal supply of credits.
code: Blackmarket uplinks on the backend were tweaked in a way consistent with other sold items.
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
was kinda hoping for a "Black Market code, Removed" but i can't say no to code improvement

**Comment by Time-Green** [comment]
> was kinda hoping for a "Black Market code, Removed" but i can't say no to code improvement
> 
> and "market datum" is really cool

Maybe if you finished merchants? 😏

**Comment by cacogen** [comment]
Nobody ever makes either of the black market things (the standing up thing or the handheld thing) and the products for sale suck shit. You shouldn't need the standing up thing and the stuff it sells should be worth making the handheld thing for (and worth remembering the feature exists).

**Comment by ArcaneMusic** [comment]
> Nobody ever makes either of the black market things (the standing up thing or the handheld thing) and the products for sale suck shit. You shouldn't need the standing up thing and the stuff it sells should be worth making the handheld thing for (and worth remembering the feature exists).

:I
Just because you don't like it doesn't mean I won't improve the backend designed for it.

**Comment by cacogen** [comment]
I like the idea of it and I'm glad you improved the code rather than removing it like Armhulen wants, but the execution of it is terrible to the point it's never used. And it could potentially be something fun if it were easier to access/people were more aware of it and how to access it and it actually had worthwhile stock.

File: code/modules/cargo/markets/market_uplink.dm [hunk 1]
```diff
@@ -152,15 +137,23 @@
...
 
 			buying = FALSE
 			selected_item = null
 
+/obj/item/market_uplink/blackmarket
+	name = "\improper Black Market Uplink"
+	desc = "An illegal black market uplink. If command wanted you to have these, they wouldn't have made it so hard to get one."
+	icon = 'icons/obj/blackmarket.dmi'
+	icon_state = "uplink"
+	accessible_markets = list(/datum/market/blackmarket) ///The original black market uplink
```
Comment (tralezab) [inline code/modules/cargo/markets/market_uplink.dm:151]: incorrect dmdoc

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -481,7 +481,10 @@
 /mob/proc/get_contents()
 
 
-//Gets ID card from a mob. If hand_firsts is TRUE hands are checked first, otherwise other slots are prioritized (for subtypes at least).
+/**
+ * Gets ID card from a mob.
+ * If hand_firsts is TRUE hands are checked first, otherwise other slots are prioritized (for subtypes at least).
+ */
```
Comment (tralezab) [inline code/modules/mob/living/living.dm:487]: you should add a line for specifically what is returned here, and heck even throw in a 
```dm
 * Arguments:
 * * hand_first - boolean that checks the hands of the mob first if TRUE
```

**Review by tralezab — CHANGES_REQUESTED** [review]
minor comment stuff

**Pull request merged**